src/jit/lower.cpp

```diff
@@ -3343,9 +3343,7 @@ Lowering::LowerArrElem(GenTree **ppTree, Compiler::fgWalkData* data)
     GenTreePtr arrObjNode = arrElem->gtArrObj;
     assert(arrObjNode->IsLocal());
 
-    GenTreePtr nextNode = arrElem->gtNext;
-    // ??? Can we have a top-level GT_ARR_ELEM ???
-    noway_assert(nextNode != nullptr);
+    GenTreePtr nextNode = arrElem;
```

Comment:
This logic definitely needs some comments for anyone to understand without spending too much of their time.